csharp/src/Ice/IceSSL/SSLEngine.cs

```diff
@@ -137,52 +123,20 @@ internal void initialize()
 
             try
             {
-                using FileStream fs = File.OpenRead(certAuthFile);
-                byte[] data = new byte[fs.Length];
-                fs.Read(data, 0, data.Length);
-
-                string strbuf = "";
+                // Try using PEM format
                 try
                 {
-                    strbuf = Encoding.UTF8.GetString(data);
+                    _caCerts.ImportFromPemFile(certAuthFile);
                 }
-                catch (Exception)
+                catch (CryptographicException)
                 {
-                    // Ignore
+                    // Expected if the file is not in PEM format.
```

Comment:
What exception do we get if the file is say a png? Do we ever issue an error like "can't load file", or do we just end up saying the the cert list is empty?